### ISSUE TITLE
gossip cluster creation fails to validate with "TLS handshake timeout" as the root error

### ISSUE THREAD

**Issue by riggtravis** [kubernetes/kops#6484]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Version 1.11.0 (git-2c2042465)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

It should be trying to launch the default `kops` preferred version of kubernetes, which looks to be `1.11.6`. Running the command `kubectl version` falls into the same patter of "no such host" then "unable to connect to the server: EOF" and finally "TLS handshake timeout"

I am about to supplement all of this information with attempting to do all of this again with the latest release of Kubernetes which is `1.13.0`. For right now I am going to finish writing up what I have to work with and then will try to reproduce this issue using this newer version of Kubernetes

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

I used terraform to create a bucket for my kops state to go into.

Then I exported that bucket information into `KOPS_STATE_STORE`

Finally I created my gossip cluster using the following command
```sh
kops create cluster vault-cluster.k8s.local --zones us-east-1a --yes
```

**5. What happened after the commands executed?**

I am unable to validate my cluster. What happens is that the cluster starts out being unable to be reached at all with an "no such host" error. This is to be expected as the cluster brings itself up, but I am including it in a full run down of what's occuring so that we can figure out what it is that's going on here.

The next thing that occurs is that my commands are unable to connect to server. The exact error given is `Unable to connect to the server: EOF`. After this comes `Unable to connect to the server: net/http: TLS handshake timeout`. This is where the process ends, and I'm not able to make any further progress from here. Even any attempt to reconfigure the cluster fails as the kops command cannot issue commands to the cluster as the kops command cannot even reach the cluster.

**6. What did you expect to happen?**

According to [this guide](http://blog.arungupta.me/gossip-kubernetes-aws-kops/) I should be able to run `kops validate cluster` and get an output somewhat like this, and then be able to interact with the cluster normally. However, what I experience is that this fails with a TLS handshake timeout and I do not have a usable cluster.

```shell
~ $ kops validate cluster
Using cluster from kubectl context: cluster.k8s.local
 
Validating cluster cluster.k8s.local
 
INSTANCE GROUPS
NAME            ROLE    MACHINETYPE    MIN    MAX    SUBNETS
master-us-east-1a    Master    m3.medium    1    1    us-east-1a
nodes            Node    t2.medium    2    2    us-east-1a
 
NODE STATUS
NAME                ROLE    READY
ip-172-20-36-52.ec2.internal    node    True
ip-172-20-38-117.ec2.internal    master    True
ip-172-20-49-179.ec2.internal    node    True
 
Your cluster cluster.k8s.local is ready
```

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

[manifest](https://gist.github.com/riggtravis/cefc7b11a9e3810090371f2e4163b3ed#file-manifest-yaml)

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

[create](https://gist.github.com/riggtravis/cefc7b11a9e3810090371f2e4163b3ed#file-create)

[validate](https://gist.github.com/riggtravis/cefc7b11a9e3810090371f2e4163b3ed#file-validate)

**9. Anything else do we need to know?**

I've been experimenting with TLS 1.3 so my current OpenSSL version is 1.1.1a. I'm also wondering if it's possible that I'm somehow set up to use a cipher suite that kops does not support. I'm about to try all of this again and explicitly instruct kops to use kubernetes 1.13.0. I'll update this issue once I see what the result of that is.

I should also note that I've really only interacted with kubernetes using minikube before. I'd gotten pretty comfortable with it, but now I have limitations that demand running it in an actual factual cloud environment. I'm quite positive this is something that I have misconfigured, but I've gone through all of the documentation I can find to try to get this resolved and looked through the old issues here on GitHub and wasn't able to find a solution that didn't involve pulling from master and running `make` against committed code from before December 2018 that the current release of kops would already have in it.

*UPDATE*: kubernets 1.13.0 fails in the exact same manner

**Comment by riggtravis** [comment]
I just tried it from a freshly installed Linux operating system to see if it behaved any differently from how my Mac behaves. It does, but still fails. Now instead of getting a TLS timeout, I'm getting an I/O timeout. When digging around for a root cause I found #2384 which suggested launching in another availability zone than us-east-1a so I tried us-east-1d and had the same i/o timeout failure

**Comment by riggtravis** [comment]
I'm experiencing what happened in #63883 which looks to have been resolved by making changes to EC2/AMI firewall rules. The reason I think this is that I have purchased a DNS and have tried using it instead of creating a gossip cluster but this also fails. I'm not sure where to start looking for resolutions to the EC2/AMI firewall rules that were described though. Everything I was able to find that was related to this set of keywords was related to security groups, however, Kops should creating its own security groups if I understand how it operates correctly. Once again I'm going to follow up by trying from the linux machine.

**Comment by riggtravis** [comment]
Okay. After a few hours I can add to my list of things that I've tried and failed to have succeed that instances that are brought up via Kops sit infinitely waiting for the joyous moment when they stop using the placeholder IP address for lookup. All of the advice I could find for this were related to performing rolling updates or at least having your master node come up successfully, but I'm totally at a loss here. A better issue reporter than me would have kep track of the issue numbers they looked over while working on this, but alas. I did not have that foresight and it's getting late. I'll likely be coming back through tomorrow to do that so that any future people who run into the problems I'm having can piece together what it is that I did so they can figure out what I've misconfigured. The fact that I've settled in on exactly one problem across both the OS X machine and the Linux machine has me 100% convinced that it's a configuration issue now, probably in my AWS account.

**Comment by riggtravis** [comment]
I've identified the issue. I opened my AWS account in 2013. So certain resources when I allocate them I get the "classic" version. In particular the load balancers that Kops uses need to be the new fangled onces that have been introduced since then. I ended up spinning off a new account for my current project and will be keeping my personal account for continued experimentation with things that aren't Kubernetes.
